public/components/integrations/components/setup_integration.tsx

```diff
@@ -301,16 +307,35 @@ export function SetupIntegrationForm({
           </EuiFormRow>
           <EuiFormRow
             label="S3 Bucket Location"
-            isInvalid={isBlurred && !config.connectionLocation.startsWith('s3://')}
+            isInvalid={isBucketBlurred && !config.connectionLocation.startsWith('s3://')}
             error={["Must be a URL starting with 's3://'."]}
           >
             <EuiFieldText
               value={config.connectionLocation}
               onChange={(event) => updateConfig({ connectionLocation: event.target.value })}
               placeholder="s3://"
-              isInvalid={isBlurred && !config.connectionLocation.startsWith('s3://')}
+              isInvalid={isBucketBlurred && !config.connectionLocation.startsWith('s3://')}
```

Comment:
do we need to do a Regex check for location and bucket checkpoint instead of a prefix check? 
https://github.com/opensearch-project/dashboards-observability/blob/cea2e854a090792f01d81301d59920a13a569650/common/constants/data_sources.ts#L39

Reply:
Might be overkill? I think UI-wise we already communicate that we need a URL and the check is enough that a user would have to be trying to break it. Not sure if we should try and play the "Write a URL validator" game ourselves unless we have a library specifically for it.